modules/ding_loan/plugins/content_types/loans.inc

```diff
@@ -170,25 +178,38 @@ function ding_loan_loans_form($form, &$form_state, $account, $loans = array()) {
 
       if (!is_object($entity) || empty($entity->reply)) {
         // Patron have this item in loans list, but opensearch can't find this
-        // record for current agencyId, so we load it separately.
-        // Here we are loading TingClientObject.
-        $entity = ding_provider_get_pseudo_entity($loan->ding_entity_id);
+        // record for current agencyId, so ask ding_provider for help:
+        $ding_entity_id = $loan->ding_entity_id;
+        $entities = ding_provider_get_pseudo_entities(array($ding_entity_id));
+        $entity = array_shift($entities);
       }
 
-      // Title link.
-      $raw_title = $entity->getTitle();
-      $uri = entity_uri('ting_object', $entity);
-      $title = l($raw_title, $uri['path']);
+     // Initialize variables so that they don't carry over if this entity can't
+     // be searched.
+     $creators = $mat_type = $cover = NULL;
+     if (is_object($entity) && !empty($entity->reply)) {
+       $raw_title = $entity->getTitle();
+
+       // Add title link.
+       $uri = entity_uri('ting_object', $entity);
+       $title = l($raw_title, $uri['path']);
+
+       // Add creator(s).
+       $creators = $entity->creators;
+
+       // Add type.
+       $mat_type = $entity->type;
 
-      // Attempt to render a cover.
-      $cover = field_view_field('ting_object', $entity, 'ting_cover', 'user_list');
+       // Attempt to render a cover.
+       $cover = field_view_field('ting_object', $entity, 'ting_cover', 'user_list');
+     }
 
       $item = array(
         '#type' => 'material_item',
         '#loan' => $loan,
         '#id' => $loan->id,
-        '#creators' => !empty($entity->creators) ? $entity->creators : NULL,
-        '#material_type' => !empty($entity->type) ? $entity->type : NULL,
+        '#creators' => isset($creators) ? $creators : NULL,
+        '#material_type' => isset($mat_type) ? $mat_type : NULL,
```

Comment:
This check is not needed since their default values are `NULL` as set on line 189.

Reply:
It violates the DRY principle, but if u're certain...
P.S. That's the greatest "no" answer I ever got.